play.pokemonshowdown.com/src/panel-popups.tsx

```diff
@@ -585,6 +586,101 @@ class AvatarsPanel extends PSRoomPanel {
 	}
 }
 
+class BattleForfeitPanel extends PSRoomPanel {
+	static readonly id = 'forfeit';
+	static readonly routes = ['forfeitbattle'];
+	static readonly location = 'semimodal-popup';
+	static readonly noURL = true;
+
+	handleForfeit = (ev: Event) => {
+		const elem = this.props.room.parentElem;
+		const roomid = (elem as HTMLInputElement)?.value as RoomID || PS.getRoom(elem)?.id || '' as RoomID;
+		const room = PS.rooms[roomid] as BattleRoom;
+
+		const closeAfter = this.base!.querySelector<HTMLInputElement>('input[name=closeroom]')?.checked;
+		room.send("/forfeit");
+		if (closeAfter) PS.leave(room.id);
+		ev.preventDefault();
+		this.close();
+	};
+
+	update = () => {
+		this.forceUpdate();
+	};
+
+	override render() {
+		const room = this.props.room;
+		const elem = room.parentElem;
+		const roomid = (elem as HTMLInputElement)?.value as RoomID || PS.getRoom(elem)?.id || '' as RoomID;
+		const battleRoom = PS.rooms[roomid] as BattleRoom;
+
+		return <PSPanelWrapper room={room} width={480}><div class="pad">
+			<form>
+				<p>Forfeiting makes you lose the battle. Are you sure?</p>
+				<p>
+					<label class="checkbox"><input type="checkbox" name="closeroom" checked={true}></input> Close after
+						forfeiting</label>
+				</p>
+				<p>
+					<button onClick={this.handleForfeit} class="button"><strong>Forfeit</strong></button>
+					<button type="button" value={battleRoom.id} data-href="replaceplayer" class="button">
+						Replace player
+					</button>
```

Comment:
```suggestion
					<button onClick={this.handleForfeit} class="button"><strong>Forfeit</strong></button> {}
					<button type="button" value={battleRoom.id} data-href="replaceplayer" class="button">
						Replace player
					</button> {}
```

This is how you make sure there are spaces between the buttons.